src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -387,7 +361,9 @@ void SolverInterfaceImpl::initializeData()
   _accessor->exportFinal();
   solverInitEvent.start(precice::syncMode);
 
-  _hasInitializedData = true;
+  _state = State::Initialized;
+
+  return _couplingScheme->getNextTimestepMaxLength();
```

Comment:
This should be the same as `dt`, correct?
Maybe you could make the `dt` `const` to make this safer.

```suggestion
  return dt;
```

Reply:
This is a small problem (and I'm not sure whether there is an easy solution): If the participant-first method is used for setting the `dt`, we have to call this function twice in initialize: First before all data exchange (because we might need `dt` already for the actions) and then after the initial data exchange (because we might have received a `dt` from first). I updated the source code a bit, but there are still two calls of `_couplingScheme->getNextTimestepMaxLength()`.

Reply:
Could you please add this information as comments?

Reply:
Done. 6865868d0061d632e3cc1846bbf1a46e60fa814f